### ISSUE TITLE
Values function does not return a list that can be expanded with ...

### ISSUE THREAD

**Issue by peteroneilljr** [hashicorp/terraform#24699]
### Terraform Version

```
$ terraform -v
Terraform v0.12.24
+ provider.aws v2.54.0
+ provider.kubernetes v1.11.1
+ provider.sdm v1.0.0
```

### Terraform Configuration Files

```hcl
resource "sdm_node" "gateway" {
  gateway {
    name           = var.sdm_gateway_name
    listen_address = "${coalesce(values(kubernetes_service.sdm_gateway_hostname.load_balancer_ingress.0)...)}:${var.sdm_port}"
  }
}
```

### Expected Behavior

The values function should return a list, and the expression "..." should expand the returned list.
 
### Actual Behavior

```
Error: Invalid expanding argument value

  on modules/terraform_aws_eks_strongdm_gateways/main.tf line 23, in resource "sdm_node" "gateway":
  23:     listen_address = "${coalesce(values(kubernetes_service.sdm_gateway_hostname.load_balancer_ingress.0)...)}:${var.sdm_port}"

The expanding argument (indicated by ...) must be of a tuple, list, or set
type.
```
### Additional Context

```
$ terraform state show module.aws_eks_sdm_gateway.kubernetes_service.sdm_gateway_hostname

# module.aws_eks_sdm_gateway.kubernetes_service.sdm_gateway_hostname:
resource "kubernetes_service" "sdm_gateway_hostname" {
    id                    = "default/aws-eks-gateway"
    load_balancer_ingress = [
        {
            hostname = "aef4801aa80e311eab110068adabce63-1474631892.us-west-2.elb.amazonaws.com"
            ip       = ""
        },
    ]
# trimmed irrelevant content
}
```
### References

Could very well be the same as #22404 but it looks like that ticket is focused on nested forloops  of lists. If it is the same feel free to merge.

**Comment by danieldreier** [comment]
@peteroneilljr thanks for reporting this! On first glance, the expected behavior you're describing here seems reasonable enough to me.

It would be helpful for me if you could simplify your example using null resources so we can run your example and reproduce the a trivial version of the behavior - it feels a little overkill to set up a k8s cluster to output a terraform list. Is that something you can help with?

Once we have that I'll get this in front of an engineer to understand why this is happening.

**Comment by peteroneilljr** [comment]
Hi Daniel, thanks for responding. I should have thought beforehand to make a simpler example. I'll keep it in mind for any future reports.

Interestingly enough creating an output with the value expanded works as expected
```hcl

locals {
  map_of_strings = { one = "", two = "string2", three = ""}
}
output "output_works" {
  value = coalesce(values(local.map_of_strings)...)
}
```

But the null_resource produces an error, although it is a bit different.
```hcl
resource "null_resource" "string_test" {
  triggers = {
    cluster_instance_ids = local.map_of_strings
  }
  provisioner "local-exec" {
    command = "echo ${coalesce(values(local.map_of_strings)...)}"
  }
}
```

output
```
Error: Incorrect attribute value type

  on null_test_bug.tf line 10, in resource "null_resource" "string_test":
  10:   triggers = {
  11:     cluster_instance_ids = local.map_of_strings
  12:   }
    |----------------
    | local.map_of_strings is object with 3 attributes

Inappropriate value for attribute "triggers": element "cluster_instance_ids":
string required.
```
Let me know if this is sufficient or if I can provide any additional info!

**Comment by danieldreier** [comment]
@peteroneilljr I might be misunderstanding the reproduction case here, but it looks to me like the last case is expected behavior. According to the [null_resource docs](https://www.terraform.io/docs/providers/null/resource.html#triggers), the `triggers` attribute accepts "A map of arbitrary strings that, when changed, will force the null resource to be replaced, re-running any associated provisioners." 

Since `local.map_of_strings` is a map, the following syntax is passing a map of maps of strings rather than a map of strings to triggers.
```
  triggers = {
    cluster_instance_ids = local.map_of_strings
  }
``` 

This is the equivalent of:
```
  triggers = {
    cluster_instance_ids = { one = "", two = "string2", three = ""}
  }
```

If I simplify that to use the map_of_strings directly, it works:
```hcl
locals {
  map_of_strings = { one = "", two = "string2", three = ""}
}
output "output_works" {
  value = coalesce(values(local.map_of_strings)...)
}

resource "null_resource" "string_test" {
  triggers = local.map_of_strings

  provisioner "local-exec" {
    command = "echo ${coalesce(values(local.map_of_strings)...)}"
  }
}
```

**Comment by apparentlymart** [comment]
Hi @peteroneilljr!

What you've described here seems to have the same root cause as #22404, which has now been fixed in our main branch ready to be included in the forthcoming Terraform v0.13.0 release. If you'd like to try it out ahead of the final release you could use v0.13.0-beta2, which is currently available. You can read more about that beta release -- including some caveats/warnings about using it for production configurations -- in [the v0.13.0 beta guide](https://github.com/hashicorp/terraform/blob/guide-v0.13-beta/README.md).

Because it took a while to fix that issue there were a few duplicates of it around which I've closed recently. I'm going to close this one in similar vein, just to consolidate the discussion into #22404. Thanks for reporting it!
